File: CPP/AbstractClassDemo.cpp
#include<iostream>
#include<cmath>
using namespace std;

class shape         //Abstract Class
{
public:
    const float PI = 3.14F;            //Constant   

    virtual void input() = 0;       //pure virtual function (abstract)
    virtual float area() = 0;       //pure virtual function (abstract)
};

class circle : public shape
{
    float rad;
public:
    void input()
    {
        cout<<"Enter radius: ";
        cin>>rad;
    }
    float area()
    {
        return PI * rad * rad;
    }
};

class rectangle : public shape
{
    float ln, br;
public:
    void input()
    {
        cout<<"Enter length: ";
        cin>>ln;
        cout<<"Enter breadth: ";
        cin>>br;
    }
    float area()
    {
        return ln * br;
    }

};

class triangle : public shape
{
    float a, b, c, s;
public:
    void input()
    {
        cout<<"Enter the value of a: ";
        cin>>a;
        cout<<"Enter the value of b: ";
        cin>>b;
        cout<<"Enter the value of c: ";
        cin>>c;
    }
    float area()            //Overriding area()
    {
        s = (a + b + c)/2;
        return (float) sqrt(s*((s-a) * (s-b) * (s-c)));      //Herons Formula
    }
};

int main()
{
    shape *s;
    s = new circle();
    s->input();                 //Runtime Polymorphism
    cout<<"The area of circle is "<<s->area()<<endl;        //Runtime Polymorphism

    s = new rectangle();
    s->input();                 //Runtime Polymorphism
    cout<<"The area of rectange is "<<s->area()<<endl;      //Runtime Polymorphism

    s = new triangle();
    s->input();                 //Runtime Polymorphism
    cout<<"The area of triangle is "<<s->area()<<endl;      //Runtime Polymorphism

    return 0;
}

File: Java/AbstractClassDemo.java
import java.util.Scanner;
abstract class Shape {                 //Abstract Class
    public final float PI = 3.14F;          //Constant declaration
 
    public abstract void input();
    public abstract float area();
    public void test() {
        System.out.println("This is a test method!");
    }
}

class Circle extends Shape {
    private float rad;
    public void input() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter radius: ");
        rad = sc.nextFloat();
        //sc.close();
    }

    public void test() {
        System.out.println("This test method is overriddin in Circle!");
    }

    public float area() {
        return PI * rad * rad;
    }
}

class Rectangle extends Shape {
    private float ln, br;
    public void input() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter length: ");
        ln = sc.nextFloat();
        System.out.print("Enter breadth: ");
        br = sc.nextFloat();
        //sc.close();
    }
    public float area() {
        return ln * br;
    }

}

class Triangle extends Shape {
    private float a, b, c, s;
    public void input() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the value of a: ");
        a = sc.nextFloat();
        System.out.print("Enter the value of b: ");
        b = sc.nextFloat();
        System.out.print("Enter the value of c: ");
        c = sc.nextFloat();
    }
    public float area() {
        return (float) Math.sqrt(s*((s-a) * (s-b) * (s-c)));
    }
}

public class AbstractClassDemo {
    public static void main(String[] args) {
        Shape s;
        s = new Circle();
        s.input();              //Runtime Polymorphism
        System.out.println("The area of circle is " + s.area());    //Runtime Polymorphism
        s.test();

        s = new Rectangle();
        s.input();              //Runtime Polymorphism
        System.out.println("The area of circle is " + s.area());    //Runtime Polymorphism

        s = new Triangle();
        s.input();              //Runtime Polymorphism
        System.out.println("The area of triangle is " + s.area());  //Runtime Polymorphism

    }
}